src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/MyAppFileAccessor.cs

```diff
@@ -65,43 +68,35 @@ private void DocData_Modifying(object sender, EventArgs e)
 
     private async Task<MyAppDocument?> TryGetMyAppFileAsync()
     {
-        if (_docData?.Data is null || !File.Exists(_absolutePath))
+        if (_docData?.Data is null)
         {
-            await _threadingService.SwitchToUIThread();
-            try
-            {
-                _docData = new DocData(_serviceProvider, _absolutePath);
-                _docData.Modifying += DocData_Modifying;
-            } 
-            catch (Exception)
+            await TaskScheduler.Default;
+            // Create My Project directory if it doesn't exist. If it does, nothing happens
+            await _storage.CreateFolderAsync("My Project");
+
+            if (!File.Exists(_absolutePath))
             {
-                // If we have reached here, it means that the file doesn't exist in that location, so try to create it
-                
-                // Create My Project directory if it doesn't exist. If it does, nothing happens
-                await _storage.CreateFolderAsync("My Project");
-                
                 // Create and write defaults to the myapp file
                 await _storage.CreateEmptyFileAsync(_absolutePath);
-                StreamWriter writer = File.AppendText(_absolutePath);
+                using StreamWriter writer = File.AppendText(_absolutePath);
                 await writer.WriteAsync(DefaultMyappFileContents);
                 await writer.FlushAsync();
-                writer.Close();
-                
-                // Now that we've created the file, we can instantiate _docData and myAppDocument.
-                _docData = new DocData(_serviceProvider, _absolutePath);
-                _docData.Modifying += DocData_Modifying;
-
-                myAppDocument = new MyAppDocument(_docData);
-                return myAppDocument;
             }
-            
-            await TaskScheduler.Default;
         }
-
-        if (myAppDocument is null)
+        
+        await _threadingService.SwitchToUIThread();
+        try
         {
+            _docData = new DocData(_serviceProvider, _absolutePath);
+            _docData.Modifying += DocData_Modifying;
             myAppDocument = new MyAppDocument(_docData);
         }
+        catch (Exception)
```

Comment:
Catching all exceptions like this is generally a bad idea, especially when the catch block is empty.

What are you actually preventing against here, and is there an exception type for that specifically?

Reply:
I meant to keep the existing logic, which catches all exceptions and returns null
https://github.com/dotnet/project-system/blob/5136a986dd9b987ab8a0824934b271998605fe8d/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/MyAppFileAccessor.cs#L60

Reply:
I just checked (git blame is your friend) and that logic comes from the comment (and PR) here: https://github.com/dotnet/project-system/pull/8252#discussion_r903592523

What Drew said is correct, it is generally poor design to 'catch everything'; for the reason is it can hide issues or even cause them. In this case, it was done to avoid when the file doesn't exist. You can ask @melytc about it, and figure out the appropriate exception for that situation.

Reply:
A `NullReferenceException` may be adequate here: we want to catch the exception raised when trying to create the `DocData`, which can happen if the `absolutePath` is not found given there's no file.

Reply:
Gotcha, thanks to the both of you 😄 I'll change to NullReferenceException